pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -493,7 +493,13 @@ def _set_parallel_devices_and_init_accelerator(self) -> None:
                     f" got `devices={self._devices_flag}` instead."
                 )
         if not self.accelerator.is_available():
-            raise MisconfigurationException(f"{self.accelerator.__class__.__qualname__} can not run on this hardware.")
+            if isinstance(self.accelerator, GPUAccelerator):
+                hardware_str = "GPU"
+            elif isinstance(self.accelerator, IPUAccelerator):
+                hardware_str = "IPU"
+            else:
+                hardware_str = "TPU"
+            raise MisconfigurationException(f"{self.accelerator.__class__.__qualname__} can not run on this hardware, " + hardware_str + "s are not available.")
```

Comment:
- the error message should show what hardware is available
- the error message should show how uses can configure this on the Trainer (e.g. `Trainer(accelerator="...")`

Reply:
updated msg